cs-amazon/src/main/java/io/cloudslang/content/amazon/services/helpers/AwsSignatureHelper.java

```diff
@@ -149,12 +152,26 @@ public String getAmazonCredentialScope(String dateStamp, String awsRegion, Strin
      * @param endpoint AWS request endpoint.
      * @return A (lowercase alphanumeric) string representing the AWS region.
      */
+    //todo does not recognize properly us-east-2.quicksight.amazonaws.com, us-west-1.queue.amazonaws.com types of hostname
     public String getAmazonRegion(String endpoint) {
-        if (isNotBlank(endpoint) && endpoint.contains(HYPHEN)) {
-            endpoint = endpoint.substring(3);
-            return endpoint.substring(START_INDEX, endpoint.indexOf(DOT_CHAR));
+        if (isBlank(endpoint)) {
+            return DEFAULT_AMAZON_REGION;
         }
-        return DEFAULT_AMAZON_REGION;
+        if (endpoint.endsWith(DOT+AMAZON_HOSTNAME)) {
```

Comment:
I would personally create a separate method to modify the **endpoint** String. Also, please consider issues raised by **codeclimate**.